src/PJ_pipeline.c

```diff
@@ -477,14 +477,35 @@ PJ *OPERATION(pipeline,0) {
         }
     }
 
-    /* Check that output units from step i are compatible with expected units in step i+1 */
+    /* Check that output units from step i are compatible with expected units in step i-1 and i+1 */
     for (i = 1; i < nsteps; i++) {
-        enum pj_io_units unit_returned = pj_right (P->opaque->pipeline[i]);
-        enum pj_io_units unit_expected = pj_left  (P->opaque->pipeline[i+1]);
+        enum pj_io_units previous_step_input  = i>1 ? pj_left  (P->opaque->pipeline[i-1]) : PJ_IO_UNITS_WHATEVER;
+        enum pj_io_units previous_step_output = i>1 ? pj_right (P->opaque->pipeline[i-1]) : PJ_IO_UNITS_WHATEVER;
+
+        enum pj_io_units current_step_input  = pj_left  (P->opaque->pipeline[i]);
+        enum pj_io_units current_step_output = pj_right (P->opaque->pipeline[i]);
+
+        enum pj_io_units next_step_input  = pj_left  (P->opaque->pipeline[i+1]);
+
+        /* If we encounter an operation that doesn't care what comes in AND out */
+        /* we override the units with the values of the neighbouring steps      */
+        if (current_step_input == PJ_IO_UNITS_WHATEVER && current_step_output == PJ_IO_UNITS_WHATEVER) {
+            if (i==1) {
+                P->opaque->pipeline[i]->left  = next_step_input;
+                P->opaque->pipeline[i]->right = next_step_input;
+            } else if (i==nsteps-1) {
+                P->opaque->pipeline[i]->left  = previous_step_output;
+                P->opaque->pipeline[i]->right = previous_step_output;
+            } else {
+                P->opaque->pipeline[i]->left  = previous_step_input;
```

Comment:
shouldn't that be previous_step_output ?

Reply:
Yes. I'll fix it right away.